scipy/optimize/_lsq/least_squares.py

```diff
@@ -667,12 +667,14 @@ def least_squares(
     First, define the function which generates the data with noise and
     outliers, define the model parameters, and generate data:
 
-    >>> def gen_data(t, a, b, c, noise=0, n_outliers=0, random_state=0):
+    >>> import numpy as np
+    >>> from numpy.random import default_rng
+    >>> rng = default_rng()
+    >>> def gen_data(t, a, b, c, noise=0., n_outliers=0, random_state=None):
```

Comment:
Put the `default_rng()` call inside the function. Rename the `random_state` argument to `seed` (for readability's sake), and do `rng = default_rng(seed)` inside of the function.